src/numerics/Func1.cpp

```diff
@@ -580,8 +1119,64 @@ Func1& newProdFunction(Func1& f1, Func1& f2)
     }
 }
 
+shared_ptr<Func1> newProdFunction(shared_ptr<Func1> f1, shared_ptr<Func1> f2)
+{
+    if (isOne(f1)) {
+        return f2;
+    }
+    if (isOne(f2)) {
+        return f1;
+    }
+    if (isZero(f1) || isZero(f2)) {
+        return shared_ptr<Func1>(new Const1(0.));
```

Comment:
Throughout here, I think this can be written like:
```suggestion
        return make_shared<Const1>(0.);
```